homeassistant/components/huawei_lte/select.py

```diff
@@ -0,0 +1,119 @@
+"""Support for Huawei LTE selects."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from functools import partial
+import logging
+from typing import Callable
+
+from huawei_lte_api.enums.net import LTEBandEnum, NetworkBandEnum
+
+from homeassistant.components.select import (
+    DOMAIN as SELECT_DOMAIN,
+    SelectEntity,
+    SelectEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HuaweiLteBaseEntityWithDevice
+from .const import DOMAIN, KEY_NET_NET_MODE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up from config entry."""
+    router = hass.data[DOMAIN].routers[config_entry.unique_id]
+    switches: list[Entity] = []
+
+    desc = SelectEntityDescription(
+        key="network_mode", name="Preferred mode", entity_category=EntityCategory.CONFIG
+    )
+    switches.append(
+        HuaweiLteBaseSelect(
+            router,
+            description=desc,
+            key=KEY_NET_NET_MODE,
+            item="NetworkMode",
+            choices={
+                "00": "4G/3G/2G",
+                "01": "2G",
+                "02": "3G",
+                "03": "4G",
+                "0301": "4G/2G",
+                "0302": "4G/3G",
+                "0201": "3G/2G",
+            },
+            setter_method=partial(
+                router.client.net.set_net_mode,
+                LTEBandEnum.ALL,
+                NetworkBandEnum.ALL,
+            ),
+        )
+    )
+
+    async_add_entities(switches, True)
+
+
+@dataclass
+class HuaweiLteBaseSelect(HuaweiLteBaseEntityWithDevice, SelectEntity):
```

Comment:
I suppose we're missing something here to set the select's name, for me it shows the device model there the UI now, instead of "Network mode" or something.

Reply:
It should be (or at least was) read earlier from the entity description object, I'll take a look soonish to see the necessary changes made.

Reply:
Looks like it wasn't, so I added `__post_init__` hook to fix this. Tests still missing.